src/Storage/Device/S3.php

```diff
@@ -386,9 +402,9 @@ private function resetResponse(): void
 
     /**
      * Get file info
-     * @return Array | false
+     * @return array | false
```

Comment:
@eldadfux is it a good idea, when there is error, returning false, if yes how to set return type of this function?

Also, I am think we need to introduce exceptions as well in this library, right now for AWS adapter, I am just returning false, anywhere there's error. This makes a lot difficult while debugging errors.

Reply:
Yes. You can throw the default PHP exception for all those cases, and keep consistent return type for better type strict codebase. If we'll need more types, we can later extend with our own custom exceptions.

Reply:
@eldadfux this is not related to this feature, so can we merge this and i'll introduce another fix PR?